rtos/mbed_boot.c

```diff
@@ -232,7 +232,7 @@ osMutexAttr_t             singleton_mutex_attr;
 #if !defined(HEAP_START)
     #if defined(__ICCARM__)
         #error "Heap should already be defined for IAR"
-    #elif defined(__CC_ARM)
+    #elif defined(__CC_ARM) || defined (__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050)
```

Comment:
You might consider wrapping `defined (__ARMCC_VERSION) && (__ARMCC_VERSION >= 6010050)` in parentheses as is done below.